pkg/gohai/memory/memory.go

```diff
@@ -6,48 +6,27 @@
 // Package memory regroups collecting information about the memory
 package memory
 
-// Memory holds memory metadata about the host
-type Memory struct {
-	// TotalBytes is the total memory for the host in byte
-	TotalBytes uint64
-	// SwapTotalBytes is the swap memory size in byte (Unix only)
-	SwapTotalBytes uint64
-}
-
-const name = "memory"
+import (
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
+)
 
-// Name returns the name of the package
-func (memory *Memory) Name() string {
-	return name
```

Comment:
The name constant is only used if Gohai is used as a binary so moving it directly in the main to make things clearer (it is hardcoded in the agent).